packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -230,7 +232,7 @@ export class Outbox {
 			});
 		}
 
-		if (this.params.containerContext.submitBatchFn === undefined) {
+		if (this.params.submitBatchFn === undefined) {
```

Comment:
I'd think we can fully remove this code path by OpGroupingManager.submitBatch() leveraging either context.submitBatchFn or context.submitFn (depending on availability) and this code not care much about presence of batch API (as its abstracted away from it).

That said, we need to ensure that we ship this feature dark first, so likely we can't couple these unrelated feature (yet).
But I think we should still re-evaluate why this code on presence of submitBatchFn - I think it was mostly to cut off old loaders, but we learned (the hard way) it's not sufficient requirement (as there are loaders that have submitBatchFn but do not support passing through compression metadata), so we deal with it indirectly, and maybe we could engage compression / chunking for old loaders as well, making code more streamlined?